web/src/lib/components/shared-components/profile-image-cropper.svelte

```diff
@@ -0,0 +1,55 @@
+<script lang="ts">
+  import { AssetResponseDto, api } from '@api';
+  import { createEventDispatcher } from 'svelte';
+  import PhotoViewer from '../asset-viewer/photo-viewer.svelte';
+  import BaseModal from './base-modal.svelte';
+  import Button from '../elements/buttons/button.svelte';
+  const dispatch = createEventDispatcher();
+  export let asset: AssetResponseDto;
+  export let publicSharedKey = '';
+  import domtoimage from 'dom-to-image';
+  import { notificationController, NotificationType } from './notification/notification';
+  import { handleError } from '$lib/utils/handle-error';
+
+  let profilePicture: HTMLDivElement;
+
+  const handleSetProfilePicture = async () => {
+    const div = profilePicture.childNodes[0];
```

Comment:
Is there a better way to target the node? Can it not be directly bound?

Reply:
The problem with directly binding the PhotoViewer component is, that it is a Svelte Component and thus does not expose the DOM Node needed.

Reply:
In which case, can you bind the element in the PhotoViewer and export it?

Reply:
I tried to bind the imgElement, which already is a variable in photo-viewer.svelte. But now i have the problem, that the imgElement binding has to passed through the BaseModal component. Could you give some advice on how to achieve this?

Reply:
Why would it need to be passed through the BaseModal component, sorry?

Reply:
The imgElement has the correct value, when logged in photo-viewer.svelte. Logging the imgElement variable in profile-image-cropper.svelte results in an undefined object.
![grafik](https://github.com/immich-app/immich/assets/22776173/3304b81c-df1e-4feb-a4f7-8122a524830f)